modules/c++/xml.lite/include/xml/lite/Speed.h

```diff
@@ -0,0 +1,341 @@
+{
+namespace lite
+{
+//=================================================================
+/*
+ * \class xml::lite::Speed
+ * \brief An efficient interface for creating XML documents
+ *
+ * This class is designed to make building XML documents simple and fast for
+ * developers. It features direct access to any element in the document and
+ * autovivification of elements and any necessary parents or siblings. 
+ * Direct access is achieved using a simple, compact syntax:
+ *
+ *  '>' shows hierarchy
+ *  '&' shows ordinality (starting with one)
+ *  example:
+ *         XML                 xml::lite::Speed Syntax
+ *  ----------------------------------------------------
+ *       <root>             |                
+ *           <foo>          | "foo"             
+ *               <bar/>     | "foo&1>bar&1" OR "foo>bar"             
+ *               <bar>      | "foo&1>bar&2" OR "foo>bar&2"
+ *                   <aye/> | "foo&1>bar&2>aye&1" OR "foo>bar&2>aye" 
+ *               </bar>     |
+ *           </foo>         |                
+ *       </root>            |                
+ *
+ *  As shown above, the name of the root element is not used when referring to
+ *  other elements in the document to prevent dull repetition. To set, modify,
+ *  or access the root element itself, use initRoot(), addRootAttribute(), and
+ *  the inherited getRootElement() method.
+ *
+ *  Apologia: The author here attempts to justify the creation of this class
+ *  and the meager but novel syntax it introduces in the presence of existing
+ *  standardized XML expression and query languages such as XPath and XQuery
+ *  (See http://www.w3.org/TR/xpath ; http://www.w3.org/TR/xpath-31 ;
+ *  http://www.w3.org/TR/xquery-3). This class seeks to solve the problem of
+ *  creating an XML document where none exists, whereas the languages mentioned
+ *  above are focused on accessing and manipulating elements in pre-existing
+ *  XML documents. In order to facilitate autovivification the syntax must
+ *  unambiguously refer to single elements only, which would mean severely
+ *  restricting allowable XPath syntax for this use. It is hoped that the
+ *  utility and shallow learning curve will excuse the introduction of a new
+ *  syntax.
+ */
+class Speed : public xml::lite::Document
+{
+  public:
+    /*!
+     * \func initRoot
+     * \brief Creates the root element of the XML document.
+     *
+     * \param[in] rootName The name of the root element of the document.
+     */
+    void initRoot(const std::string &rootName);
+
+    /*!
+     * \func addRootAttribute
+     * \brief Adds an XML attribute to the root element of the document.
+     *
+     * \param[in] attrname A string that is the name of the new attribute.
+     * \param[in] value    Value of the attribute.
+     *                     If value is not a string it will be turned into one internally.
+     */
+    template <typename T>
+      void addRootAttribute(const std::string &attrname, T value);
```

Comment:
I agree with you and at the same time hope that attribute strings in XML are not huge. However, allowing this to play nicely in those inscrutable situations takes little effort, so I'll put this in.

As for the two spaces, I think vim was over-eager on the indentation. =]